src/main/java/com/labsynch/labseer/chemclasses/jchem/ChemStructureServiceJChemImpl.java

```diff
@@ -1465,8 +1444,25 @@ public boolean dropJChemTable(String tableName) {
 		return tableDropped;
 	}
 
+	public String getJchemStructureTableFromStructureType(StructureType structureType) {
```

Comment:
This function is very confusing to me. Don't you have a `structureType.table` attribute for this same purpose?

Reply:
Not really no.  Its up to the subclasses to implement these functions in anyway they like, the thing I added was to be able to look up the base table for each of the entities in case they needed it.  The structure tables are different per implementation.  e.g. For the `StructureType.PARENT` - Indigo Impl uses the Bingo indexed `PARENT` table, rdkit they are stored in `RDKIT_STRUCTURE` and for Jchem they are stored in `PARENT_STRUCTURE` table.  It would be pretty natural to subclass the enums themselves but I didn't know how to do that so I just created these switch statments.

Reply:
Ah.... `StructureType.table` is the "plain table" which would be `parent` or `salt_form` or `salt`, not the "structure table".